tests/lax_numpy_test.py

```diff
@@ -4709,6 +4709,80 @@ def assertListOfArraysEqual(xs, ys):
                         atol=atol,
                         rtol=rtol)
 
+  def testR_(self):
+    a = np.arange(6).reshape((2,3))
+    self.assertArraysEqual(np.r_[np.array([1,2,3]), 0, 0, np.array([4,5,6])],
+                           jnp.r_[np.array([1,2,3]), 0, 0, np.array([4,5,6])])
+    self.assertArraysEqual(np.r_['-1', a, a], jnp.r_['-1', a, a])
+    self.assertArraysEqual(np.r_['0,2', [1,2,3], [4,5,6]], jnp.r_['0,2', [1,2,3], [4,5,6]])
+    self.assertArraysEqual(np.r_['0,2,0', [1,2,3], [4,5,6]], jnp.r_['0,2,0', [1,2,3], [4,5,6]])
+    self.assertArraysEqual(np.r_['1,2,0', [1,2,3], [4,5,6]], jnp.r_['1,2,0', [1,2,3], [4,5,6]])
+    # negative 1d axis start
+    self.assertArraysEqual(np.r_['0,4,-1', [1,2,3], [4,5,6]], jnp.r_['0,4,-1', [1,2,3], [4,5,6]])
+    self.assertArraysEqual(np.r_['0,4,-2', [1,2,3], [4,5,6]], jnp.r_['0,4,-2', [1,2,3], [4,5,6]])
+
+    # matrix directives
+    with warnings.catch_warnings():
+      warnings.filterwarnings("ignore", category=PendingDeprecationWarning)
```

Comment:
What warning does this catch? If numpy is going to be deprecating something, I think we should avoid adding that deprecated API to JAX.

Same below.